src/test/java/org/shikshalokam/backend/scp/TestScpPermissionCRUDOperations.java

```diff
@@ -175,33 +176,65 @@ private Response createPermission(String prefix, String module, String requestTy
         return response;
     }
 
-    @Test(description = "Verifies the getPermissions API for a valid user.")
+    @Test(priority = 7 , description = "Verifies the getPermissions API for a valid user.")
     public void testGetPermissionsWithValidPayload() {
         logger.info("Started calling the GetPermissions API with valid token.");
 
         Response response = getPermissions(true);
         response.prettyPrint();
 
         // Validate response code
-        Assert.assertEquals(response.getStatusCode(), 200, "Failed to fetch permissions.");
+        assertEquals(response.getStatusCode(), 200, "Failed to fetch permissions.");
         logger.info("Permissions retrieved successfully.");
     }
 
-    @Test(description = "Verifies the getPermissions API with an invalid token.")
+    @Test(priority = 8 , description = "Verifies the getPermissions API with an invalid token.")
     public void testGetPermissionsApiWithInvalidToken() {
         logger.info("Started calling the GetPermissions API with an invalid token.");
 
         Response response = getPermissions(false);
 
         // Validate response code for invalid token
-        Assert.assertEquals(response.getStatusCode(), 401, "Expected 401 Unauthorized error.");
+        assertEquals(response.getStatusCode(), 401, "Expected 401 Unauthorized error.");
         logger.info("Received expected error for invalid token.");
     }
+    @Test(priority = 9 , description = "Verifies the functionality of deleting user's permission by ID.")
+    public void testDeletePermissionById() {
+        logger.info("Started calling the DeletePermission API with valid ID.");
 
+        Response response = deletePermissionById(createdId);
 
+        // Log the status code and response body
+        int statusCode = response.getStatusCode();
+        response.prettyPrint();
+
+        // Validate response code is 200 for a successful deletion
+        Assert.assertEquals(statusCode, 202, "Status code should be 200 for successful deletion");
+
+        logger.info("Ended calling the DeletePermission API with valid ID.");
+    }
+    @Test(priority = 10, description = "Verifies the functionality of retrieving permissions list based on a specific role.")
+    public void testGetListPermissionsByRole() {
+        logger.info("Started calling the GetListPermissionsByRole API with a valid role.");
+
+        // Call getListPermissionsByRole with a specific role
+        Response response = getListPermissionsByRole("admin");  // Replace "admin" with the actual role to test
```

Comment:
Do we have role called "admin" ?
if hardcode like this, how it will run in prod .
Just see how we can achieve in all the environments